### PR TITLE
Topological sort implementation. With tests.

### PR DESCRIPTION
**Pull request by gszjulcsi**
I have implemented topological sorting from the todo list (https://github.com/thcyron/graphs/issues/1)
Please review it, and merge if you like it.

### CODE REVIEW

File: topological_sort.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	for len(inEdges) > 0 {
+		topClass := []Vertex{}
+		for v, inDegree := range inEdges {
+			if inDegree == 0 {
+				topClass = append(topClass, v)
+				topologicalClasses[v] = tClass
+			}
+		}
+		if len(topClass) == 0 {
+			fmt.Println("This graph is not a DAG. No topological order exists.")
```
Comment (thcyron) [inline topological_sort.go]: I would prefer to return an exported error like `ErrNoDAG` here so caller’s of `TopologicalSort` can check for that specific error.

File: topological_sort_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		t.Error("Node 2 has wrong class")
+	}
+	if topClasses[3] != 1 {
+		t.Error("Node 3 has wrong class")
+	}
+
+	// check if the topological ordering is correct	e := topOrder.Front()
+	e := topOrder.Front()
+	if e.Value.(int) != 1 {
+		t.Error("First node shuld be: 1")
```
Comment (thcyron) [inline topological_sort_test.go]: Minor typo.

Comment (gszjulcsi) [inline topological_sort_test.go]: Fixed, thanks!

**Comment by gszjulcsi** [comment]
I added a custom error type, it is not very sophisticated though.
On the other hand, I realised that maybe it's more safe that in case of error we return empty containers instead of nil. 
What do you think about it? 
I can change it back to the original version, if you prefer that design.

**Comment by thcyron** [comment]
I had an error variable [like in the database/sql package](http://golang.org/pkg/database/sql/#pkg-variables) in mind to make it consistent with the standard library.

```go
var ErrNoDAG = errors.New("graphs: graph is not a DAG")
```

Clients can then check with:

```go
order, classes, err := graphs.TopologicalSort(g)
if err == graphs.ErrNoDAG {
    // ...
}
```

Otherwise it looks good to me.

**Pull request merged**